SEImplementation/SEImplementation/Configuration/DetectionImageConfig.h

```diff
@@ -62,10 +63,16 @@ class DetectionImageConfig : public Euclid::Configuration::Configuration {
   // we still need to know what it was to adjust the weight map
   double getOriginalFluxScale() const { return m_flux_scale; }
 
+  // Get the detection image source
+  std::shared_ptr<FitsImageSource<DetectionImage::PixelType>> getImageSource() {
```

Comment:
I guess I should selected that line...

Reply:
> Maybe we could have an abstract class for sources that can return or generate FITS compatible headers? Or maybe just return a generic ImageSource?

Sounds like a good idea. However returning an `ImageSource` would imply adding a virtual method to it that allows getting the headers, and that would be cumbersome for all the image sources we have around (scaled, buffered, etc). I like having a different intermediate type better. Shall we call it `FileImageSource` or something like that?

And maybe the method ought to be `getMetadata`, `getHeader` and return a list of key/value/comment or something like that.

The output being fits-like is the business of the LDAC output format, I'd say.

Reply:
Maybe ImageSourceWithMetaData since it doesn't have to come from a file? I think that would be great! I could also remove the dependency from the WCS coordinate system since it's really the header metadata we need not the file itself.

Reply:
Added the new class. I have not modified the WCS part, though. I think that'd be outside the scope of this merge?

Reply:
Looks good to me!